packages/reason-css-parser/ppx/Generate.re

```diff
@@ -636,5 +658,175 @@ module Make = (Ast_builder: Ppxlib.Ast_builder.S) => {
+        [%stri let build_variant = (~loc, name, args) => Ast_helper.Exp.variant(~loc, name, args) ],
+        [%stri let txt = (~loc, txt) => {Location.loc: loc, txt}],
+        [%stri let list_to_longident = vars => vars |> String.concat(".") |> Lexing.from_string |> Parse.longident],
+        [%stri let render_interpolation : (~loc: Location.t, Types.interpolation) => Parsetree.expression = (~loc, name) => list_to_longident(name) |> txt(~loc) |> Ast_helper.Exp.ident],
+				[%stri let render_string : (~loc: Location.t, string) => Parsetree.expression  = (~loc, string) => Helper.Const.string(~quotation_delimiter="js", string) |> Ast_helper.Exp.constant(~loc)],
+				[%stri let render_integer : (~loc: Location.t, Types.integer) => Parsetree.expression  = (~loc, int) => Helper.Const.int(int) |> Helper.Exp.constant(~loc)],
+				[%stri let render_number : (~loc: Location.t, Types.number) => Parsetree.expression = (~loc, number) => Helper.Const.float(number |> string_of_float) |> Helper.Exp.constant(~loc)],
+        [%stri let render_length : (~loc: Location.t, Types.length) => Parsetree.expression  = {
+              (~loc) => fun
+              | `Cap(n) => build_variant(~loc, "cap", Some(render_number(~loc, n)))
+              | `Ch(n) => build_variant(~loc, "ch", Some(render_number(~loc, n)))
+              | `Cm(n) => build_variant(~loc, "cm", Some(render_number(~loc, n)))
+              | `Em(n) => build_variant(~loc, "em", Some(render_number(~loc, n)))
+              | `Ex(n) => build_variant(~loc, "ex", Some(render_number(~loc, n)))
+              | `Ic(n) => build_variant(~loc, "ic", Some(render_number(~loc, n)))
+              | `In(n) => build_variant(~loc, "In", Some(render_number(~loc, n)))
+              | `Lh(n) => build_variant(~loc, "lh", Some(render_number(~loc, n)))
+              | `Mm(n) => build_variant(~loc, "mm", Some(render_number(~loc, n)))
+              | `Pc(n) => build_variant(~loc, "pc", Some(render_number(~loc, n)))
+              | `Pt(n) => build_variant(~loc, "pt", Some(render_integer(~loc, n |> int_of_float)))
+              | `Px(n) => build_variant(~loc, "pxFloat", Some(render_number(~loc, n)))
+              | `Q(n) => build_variant(~loc, "Q", Some(render_number(~loc, n)))
+              | `Rem(n) => build_variant(~loc, "em", Some(render_number(~loc, n)))
+              | `Rlh(n) => build_variant(~loc, "rlh", Some(render_number(~loc, n)))
+              | `Vb(n) => build_variant(~loc, "vb", Some(render_number(~loc, n)))
+              | `Vh(n) => build_variant(~loc, "vh", Some(render_number(~loc, n)))
+              | `Vi(n) => build_variant(~loc, "vi", Some(render_number(~loc, n)))
+              | `Vmax(n) => build_variant(~loc, "vmax", Some(render_number(~loc, n)))
+              | `Vmin(n) => build_variant(~loc, "vmin", Some(render_number(~loc, n)))
+              | `Vw(n) => build_variant(~loc, "vw", Some(render_number(~loc, n)))
+              | `Zero => build_variant(~loc, "zero", None);
+            }],
```

Comment:
Why those standard printers are different from the previous implementation?

Reply:
Because we dont have access to the variants parameter `n`; 

before we would do like this:

```\`Rem(n) -> [%expr \`rem([%e render_number(~loc, n))]```

where n would be parsed before being wrapped into a `pexp_variant`;

`build_variant` is a function that is being generated to be called by the generated code so we can achieve the same as before

`let build_variant = (~loc, name, args) => Ast_helper.Exp.variant(~loc, name, args) `

So that the same process occurs (render the arguments and then wrap it into a `pexp_variant`

Reply:
Right, I prefer to remove the build_variant function and keep them as close as before.
```
| `Zero => [%expr `zero]
| `Vmin(n) => [%expr `vmin([%e render_number(~loc, n)]]
```